### PR TITLE
Closes #10386: Add AutofillUseCases for common Android Autofill tasks.

### LINKED ISSUE #10386: Android Autofill: Add helper for enabling/disabling/checking availability
It would be nice if we could provide a helper in the component to make those actions just a one liner in app code.

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/autofill/src/main/res/layout/mozac_feature_autofill_preference.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<androidx.appcompat.widget.SwitchCompat xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/switch_widget"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent"
+    android:clickable="false"
```
Comment (jonalmeida) [inline components/feature/autofill/src/main/res/layout/mozac_feature_autofill_preference.xml:9]: Should this be true or was it meant to be changed programmatically?

Comment (jonalmeida) [reply components/feature/autofill/src/main/res/layout/mozac_feature_autofill_preference.xml:9]: Same for `focusable`?

**Review by jonalmeida — APPROVED** [review]

File: components/feature/autofill/src/main/res/layout/mozac_feature_autofill_preference.xml (hunk 1 above)
Comment (pocmo) [reply components/feature/autofill/src/main/res/layout/mozac_feature_autofill_preference.xml:9]: Yeah, this was on purpose, we do not want this individual view be clickable/focusable, but the whole preference entry.

**Comment by pocmo** [comment]
`AppLinksFeatureTest. feature observes app intents when started` - This seems to be an intermittent.

**Pull request merged**